comp/core/telemetry/telemetry.go

```diff
@@ -23,12 +24,16 @@ var (
 	registry = newRegistry()
 	provider = newProvider(registry)
 	mutex    = sync.Mutex{}
+
+	defaultRegistry = prometheus.NewRegistry()
```

Comment:
Looking forward to when the above TODO is completed, and these globals get moved into the telemetryImpl struct. But for now keeping this pattern is fine.

To verify, the other `registry` field calls: 
```
reg.MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
reg.MustRegister(collectors.NewGoCollector())
```
Whereas this one `defaultRegistery` does not register these collectors. That's intended behavior?

Reply:
Very much so, we only want a small set of hand-picked metrics by default. Go and process telemetry should remain an opt-in.